include/geode/geometry/detail/aabb_impl.h

```diff
@@ -391,4 +409,114 @@ namespace geode
+        {
+            return;
+        }
+
+        // Leaf case
+        if( is_leaf( element_begin, element_end ) )
+        {
+            // @todo Check if the box is not intersecting itself
+            action( mapping_morton_[element_begin] );
+            return;
+        }
+
+        index_t box_middle, child_left, child_right;
+        get_recursive_iterators( node_index, element_begin, element_end,
+            box_middle, child_left, child_right );
+
+        ray_trace_intersect_recursive< ACTION >(
+            ray, child_left, element_begin, box_middle, action );
+        ray_trace_intersect_recursive< ACTION >(
+            ray, child_right, box_middle, element_end, action );
+    }
+
+    template < index_t dimension >
+    bool ray_box_intersection( const InfiniteLine< dimension >& ray,
+        const BoundingBox< dimension >& box )
+    {
+        const auto box_center = ( box.min() + box.max() ) / 2.;
+        const auto box_extent = ( box.max() - box.min() ) / 2.;
+
+        // Transform the ray to the aligned-box coordinate system.
+        const auto ray_origin = ray.origin() - box_center;
+        const auto rayDirection = ray.direction();
+
+        for( const auto i : geode::Range{ dimension } )
+        {
+            if( std::fabs( ray_origin.value( i ) ) > box_extent.value( i )
+                && ray_origin.value( i ) * rayDirection.value( i ) >= 0. )
+            {
+                return false;
+            }
+        }
+        return line_box_intersection( ray_origin, rayDirection, box_extent );
+    }
+
+    template < index_t dimension >
+    bool line_box_intersection( const Point< dimension >& ray_origin,
+        const geode::Vector< dimension >& rayDirection,
+        const Point< dimension >& box_extent );
+
+    template <>
+    bool line_box_intersection( const Point3D& ray_origin,
+        const geode::Vector3D& rayDirection,
+        const Point3D& box_extent )
+    {
+        const auto WxD = geode::Vector3D{ rayDirection }.cross( ray_origin );
+        absl::FixedArray< double > absWdU{ std::fabs( rayDirection.value( 0 ) ),
+            std::fabs( rayDirection.value( 1 ) ),
+            std::fabs( rayDirection.value( 2 ) ) };
+
+        if( std::fabs( WxD.value( 0 ) )
+            > box_extent.value( 1 ) * absWdU[2]
+                  + box_extent.value( 2 ) * absWdU[1] )
+        {
+            return false;
+        }
+
+        if( std::fabs( WxD.value( 1 ) )
+            > box_extent.value( 0 ) * absWdU[2]
+                  + box_extent.value( 2 ) * absWdU[0] )
+        {
+            return false;
+        }
+
+        if( std::fabs( WxD.value( 2 ) )
+            > box_extent.value( 0 ) * absWdU[1]
+                  + box_extent.value( 1 ) * absWdU[0] )
+        {
+            return false;
+        }
+
+        return true;
+    }
+    template <>
+    bool line_box_intersection( const Point2D& ray_origin,
+        const geode::Vector2D& rayDirection,
```

Comment:
```suggestion
        const geode::Vector2D& ray_direction,
```
Why not using directly the Ray object? Instead of passing two arguments.